lightproc/examples/proc_run.rs

```diff
@@ -0,0 +1,54 @@
+//! A function that runs a future to completion on a dedicated thread.
+
+use std::future::Future;
+use std::sync::Arc;
+use std::thread;
+
+use crossbeam::channel;
+use futures::executor;
+use lightproc::prelude::*;
+
+fn spawn_on_thread<F, R>(fut: F) -> ProcHandle<R>
+where
+    F: Future<Output = R> + Send + 'static,
+    R: Send + 'static,
+{
+    let (sender, receiver) = channel::unbounded();
+    let sender = Arc::new(sender);
+    let s = Arc::downgrade(&sender);
+
+    let future = async move {
+        let _ = sender;
+        fut.await
+    };
+
+    let schedule = move |t| s.upgrade().unwrap().send(t).unwrap();
+    let (proc, handle) = LightProc::build(
+        future,
+        schedule,
+        ProcStack::default()
+            .with_pid(1)
+            .with_before_start(|| {
+                println!("Before start");
+            })
+            .with_after_complete(|| {
+                println!("After complete");
+            }),
+    );
+
+    proc.schedule();
+
+    thread::spawn(move || {
+        for proc in receiver {
+            proc.run();
+        }
+    });
```

Comment:
Maybe use `QUEUE` from `lightproc/examples/proc_panic.rs` instead? :)

Reply:
Let's keep it like that. To show direct thread mapping.